arduino/wrappers/Utils.cpp

```diff
@@ -0,0 +1,153 @@
+/* This is the implementation of wrapper classes in header.
```

Comment:
Is wrapper like a mock ? Or am I misunderstanding something?

Reply:
We used wrappers to be able to call the additional functions in related libiraires for the Smartcar, Arduino, and so on. Google Mock is related to dependencies.  I believe the link below will help you further :)

**Google Mock:**
https://chromium.googlesource.com/external/github.com/google/googletest/+/refs/tags/release-1.8.0/googlemock/docs/ForDummies.md